### PR TITLE
feat(Mathlib/Topology/StoneCech): Add theorem disjoint_closure_of_disjoint_closed

### PR DESCRIPTION
**Pull request by mapehe**
Two disjoint closed subsets of a normal space X have disjoint closures in the Stone-Cech compactification of X.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/StoneCech.lean [hunk 1]
```diff
@@ -331,4 +332,44 @@ instance StoneCech.compactSpace : CompactSpace (StoneCech α) :=
...
+      apply_rules [IsClosed.preimage, continuous_stoneCechExtend,
+        IsCompact.isClosed, isCompact_singleton, continuous_def.2]
+      simp only [preimage_id', imp_self, forall_const]
+    rw [←closed_image]
+    apply closure_mono
+    rw [←stoneCechExtend_extends hg, preimage_comp, image_subset_iff]
+  have closureSub: ∀ (x : ℝ), ∀ (u : Set α), (h0: (∀ (o : α), (o ∈ u)  → f o = x))
+      → (h : (x ∈ Set.Icc (0 : ℝ) 1)) → closure (stoneCechUnit '' u) ⊆ uu ⁻¹' {⟨x, h⟩} := by
+    intros x u hu xicc _
+    apply Subset.trans _ (subs x xicc)
```
Comment (mapehe) [inline Mathlib/Topology/StoneCech.lean:359]: https://github.com/leanprover-community/mathlib4/pull/7750#discussion_r1364483179

File: Mathlib/Topology/StoneCech.lean [hunk 2]
```diff
@@ -331,4 +332,44 @@ instance StoneCech.compactSpace : CompactSpace (StoneCech α) :=
...
+    apply closure_mono
+    apply image_subset
+    intros a ha
+    rw [mem_preimage, mem_singleton_iff, ULift.up_inj, Subtype.mk.injEq]
+    apply hu
+    exact ha
+  have subS : closure (stoneCechUnit '' s) ⊆ uu ⁻¹' {⟨0, by simp⟩} := by
+    apply_rules [closureSub 0, hfs]
```
Comment (mapehe) [inline Mathlib/Topology/StoneCech.lean:367]: https://github.com/leanprover-community/mathlib4/pull/7750#discussion_r1364483950

File: Mathlib/Topology/StoneCech.lean (hunk 1 above)
Comment (mapehe) [reply Mathlib/Topology/StoneCech.lean:359]: I'm afraid I don't really understand these comments involving `show_term` @alexjbest, could you elaborate a bit on how that tactic is used?

File: Mathlib/Topology/StoneCech.lean [hunk 3]
```diff
@@ -331,4 +332,44 @@ instance StoneCech.compactSpace : CompactSpace (StoneCech α) :=
...
+  have closureSub: ∀ (x : ℝ), ∀ (u : Set α), (h0: (∀ (o : α), (o ∈ u)  → f o = x))
+      → (h : (x ∈ Set.Icc (0 : ℝ) 1)) → closure (stoneCechUnit '' u) ⊆ uu ⁻¹' {⟨x, h⟩} := by
+    intros x u hu xicc _
+    apply Subset.trans _ (subs x xicc)
+    apply closure_mono
+    apply image_subset
+    intros a ha
+    rw [mem_preimage, mem_singleton_iff, ULift.up_inj, Subtype.mk.injEq]
+    apply hu
+    exact ha
```
Comment (mapehe) [inline Mathlib/Topology/StoneCech.lean:365]: https://github.com/leanprover-community/mathlib4/pull/7750#discussion_r1364483470

This one I couldn't get to work:
```
application type mismatch
  hu ha
argument
  ha
has type
  a ∈ u : Prop
but is expected to have type
  α : Type u
```

Comment (alexjbest) [reply Mathlib/Topology/StoneCech.lean:365]: Ah you might need some underscores like `exact hu _ ha`

File: Mathlib/Topology/StoneCech.lean (hunk 1 above)
Comment (mapehe) [reply Mathlib/Topology/StoneCech.lean:359]: Nice, that's a really nice tip!

**Comment by digama0** [comment]
Note: I have pushed an update to the lean toolchain because this PR was on a [buggy version](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Potential.20data.20loss.20from.20.60lake.20clean.60.20with.204.2E2.2E0-rc2.2F3/near/397875701) of the toolchain. **WARNING**: checking out old commits of this PR using v4.2.0-rc2 or v4.2.0-rc3 can cause `lake clean` to delete your mathlib folder! If you need to do so, make sure to delete `lakefile.olean` manually before running any `lake` commands.

**Comment by mapehe** [comment]
Based on feedback from another PR, I'm closing this at least for now.

**Pull request closed without merging**